docs/CMakeLists.txt

```diff
@@ -8,28 +8,48 @@
 #
 
 # qdoc toolchain
-# TODO: some of this probably should be upstreamed to Qt's cmake files...
-# TODO replace with Qt::qhelpgenerator for Qt >= 5.7.1
-find_program(QHELPGEN_EXECUTABLE qhelpgenerator HINTS ${QT_INSTALL_BINS})
-find_program(QDOC_EXECUTABLE qdoc HINTS ${QT_INSTALL_BINS})
-find_program(QTATTRIBUTIONSSCANNER_EXECUTABLE qtattributionsscanner HINTS ${QT_INSTALL_BINS})
+find_package(Qt${Qt_VERSION_MAJOR} NO_MODULE QUIET OPTIONAL_COMPONENTS Help)
+if(Qt_VERSION_MAJOR EQUAL 5)
+    find_package(Qt${Qt_VERSION_MAJOR} NO_MODULE QUIET OPTIONAL_COMPONENTS AttributionsScannerTools)
+elseif(Qt_VERSION_MAJOR GREATER_EQUAL 6)
+    find_package(Qt${Qt_VERSION_MAJOR} NO_MODULE QUIET OPTIONAL_COMPONENTS ToolsTools)
+endif()
+
+if (TARGET Qt::qdoc)
+    set(QDOC_EXECUTABLE "$<TARGET_FILE:Qt::qdoc>")
+else()
+    find_program(QDOC_EXECUTABLE qdoc HINTS ${QT_INSTALL_BINS})
+endif()
+
+if (TARGET Qt::qhelpgenerator)
+    # Required for Doxygen
+    get_target_property(QHELPGEN_EXECUTABLE Qt::qhelpgenerator IMPORTED_LOCATION)
+else()
+    find_program(QHELPGEN_EXECUTABLE qhelpgenerator HINTS ${QT_INSTALL_BINS})
+endif()
+
+if (TARGET Qt::qtattributionsscanner)
+    set(QTATTRIBUTIONSSCANNER_EXECUTABLE Qt::qtattributionsscanner)
+else()
+    find_program(QTATTRIBUTIONSSCANNER_EXECUTABLE qtattributionsscanner HINTS ${QT_INSTALL_BINS})
+endif()
```

Comment:
Three different tools, three different ways of consuming the target. Because:

1. `qdoc` is used in `add_custom_command()` as an argument to `${CMAKE_COMMAND} -E env ...` commands, meaning that the actual path to the binary has to be supplied. `$<TARGET_FILE:Qt::qdoc>` is the recommended method of doing so, according to [the CMake docs for `add_custom_command`]()>
   > Arguments to COMMAND may use [generator expressions](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#manual:cmake-generator-expressions(7)). Use the [TARGET_FILE](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#genex:TARGET_FILE) generator expression to refer to the location of a target later in the command line (i.e. as a command argument rather than as the command to execute).

2. `Qt::qtattributionsscanner` is used directly as the `COMMAND` supplied to `add_custom_command()`, which means that the target name can be used as the command.

3. The `Qt::qhelpgenerator` binary path needs to be supplied directly in the `QHELPGEN_EXECUTABLE` variable, as it'll be substituted into the generated Doxyfile for the API doc generation.